supercommand_test.go

```diff
@@ -230,23 +234,28 @@ func (s *SuperCommandSuite) TestVersionNotProvided(c *gc.C) {
 		Purpose: "to be purposeful",
 		Doc:     "doc\nblah\ndoc",
 	})
-	ctx := cmdtesting.Context(c)
 
 	// juju version
-	baselineCode := cmd.Main(jc, ctx, []string{"version"})
+	baselineCode := cmd.Main(jc, s.ctx, []string{"version"})
 	c.Check(baselineCode, gc.Not(gc.Equals), 0)
-	c.Assert(c.GetTestLog(), gc.Matches, `(?s).*unrecognized command: jujutest version.*`)
+	c.Assert(cmdtesting.Stderr(s.ctx), gc.Equals, "ERROR unrecognized command: jujutest version\n")
+
+	s.TearDownTest(c)
+	s.SetUpTest(c)
```

Comment:
Makes me think this test should be split into different tests. It's obviously doing too much.